src/pyclass.rs

```diff
@@ -46,14 +75,13 @@ pub trait PyClassAlloc: PyTypeInfo + Sized {
     unsafe fn dealloc(py: Python, self_: *mut Self::Layout) {
         (*self_).py_drop(py);
         let obj = PyAny::from_borrowed_ptr_or_panic(py, self_ as _);
-        if Self::is_exact_instance(obj) && ffi::PyObject_CallFinalizerFromDealloc(obj.as_ptr()) < 0
-        {
-            // tp_finalize resurrected.
-            return;
-        }
 
-        match (*ffi::Py_TYPE(obj.as_ptr())).tp_free {
-            Some(free) => free(obj.as_ptr() as *mut c_void),
+        match get_type_free(ffi::Py_TYPE(obj.as_ptr())) {
+            Some(free) => {
+                let ty = ffi::Py_TYPE(obj.as_ptr());
+                free(obj.as_ptr() as *mut c_void);
+                ffi::Py_DECREF(ty as *mut ffi::PyObject);
```

Comment:
I'm sorry for my confusing comment, but this `DECREF` should be always called.

Reply:
`tp_free_fallback` already does it.